modin/experimental/core/execution/native/implementations/hdk_on_native/calcite_serializer.py

```diff
@@ -325,6 +338,22 @@ def serialize_literal(self, literal):
                 "type_scale": -2147483648,
                 "type_precision": 1,
             }
+        if isinstance(val, np.datetime64):
+            unit = np.datetime_data(val)[0]
+            precision = self._TIMESTAMP_PRECISION.get(unit, None)
+            if precision is not None:
+                if precision == 0 and unit != "s":
+                    val = val.astype("datetime64[s]")
```

Comment:
can this ever be lower-than-second, e.g. M8[D]?  if so, then .astype can silently overflow